pkg/gardenlet/controller/backupbucket/backup_bucket_actuator.go

```diff
@@ -269,12 +269,11 @@ func (a *actuator) waitUntilBackupBucketExtensionReconciled(ctx context.Context)
 			}
 
 			if generatedSecretRef != nil || backupBucket.Status.ProviderStatus != nil {
-				_, err := controllerutil.CreateOrUpdate(ctx, a.gardenClient.Client(), a.backupBucket, func() error {
+				return kutil.TryUpdateStatus(ctx, kretry.DefaultBackoff, a.gardenClient.Client(), a.backupBucket, func() error {
```

Comment:
Can you adapt to use the `DirectClient`? I see that some other parties (the progressReporter for example) that update the status use the `DirectClient`.

```suggestion
				return kutil.TryUpdateStatus(ctx, kretry.DefaultBackoff, a.gardenClient.DirectClient(), a.backupBucket, func() error {
```